modules/dataplex-datascan/main.tf

```diff
@@ -0,0 +1,213 @@
+/**
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+locals {
+  prefix                      = var.prefix == null ? "" : "${var.prefix}-"
+  _file_data_quality_spec_raw = var.data_quality_spec_file != null ? yamldecode(file(var.data_quality_spec_file.path)) : tomap({})
+  _file_data_quality_spec = var.data_quality_spec_file == null ? null : {
+    sampling_percent = try(local._file_data_quality_spec_raw.samplingPercent, local._file_data_quality_spec_raw.sampling_percent, null)
+    row_filter       = try(local._file_data_quality_spec_raw.rowFilter, local._file_data_quality_spec_raw.row_filter, null)
+    rules = [
```

Comment:
Nit: I would move the rules block into its own local, and use a `rules.tf` file for it to keep this file more readable.

Reply:
Thanks for your review Ludo. Will submit a quick change to move rules out to a separate file, and then merge.